src/base/bittorrent/torrenthandle.cpp

```diff
@@ -487,6 +487,7 @@ bool TorrentHandle::needSaveResumeData() const
 
 void TorrentHandle::saveResumeData()
 {
+    updateStatus(); // to update queue_position, see discussion in PR #6154
```

Comment:
Is this sane? We will lock up everytime we periodically save fastresume during runtime.